### PR TITLE
Update Mesa to 21.3.9

### PR DESCRIPTION
**Pull request by Toasterson**
As discussed on Sunday, I was trying to get this working cleanly but I can't get pkgdepend to find libraries inside /usr/lib/64, any suggestions?

### CODE REVIEW

**Comment by mtelka** [comment]
Hm, `meson`.  Maybe this is related: https://github.com/mesonbuild/meson/issues/6541

**Comment by AndWac** [comment]
╰─➤  ldd libEGL.so.1.0.0
	libssp.so.0 =>	 /usr/gcc/13/lib/amd64/libssp.so.0
	libgbm.so.1 =>	 /usr/lib/64/libgbm.so.1
	libglapi.so.0 =>	 /usr/lib/64/libglapi.so.0
	libexpat.so.1 =>	 /usr/lib/64/libexpat.so.1
	libX11-xcb.so.1 =>	 /usr/lib/64/libX11-xcb.so.1
	libxcb.so.1 =>	 /usr/lib/64/libxcb.so.1
	libxcb-dri2.so.0 =>	 /usr/lib/64/libxcb-dri2.so.0
	libxcb-xfixes.so.0 =>	 /usr/lib/64/libxcb-xfixes.so.0
	libdrm.so.2 =>	 (file not found)
^^^^

**Comment by Toasterson** [comment]
Ok, I got it to the point where ldd finds all libs on my system pkgdepend still complains about libs in `/usr/lib/64` for some reason but that might be a bug on my system.

**Comment by Toasterson** [comment]
So I've somehow managed to fix my pkgdepend finally... I'll merge in geoff's changes, cleanup the patches and sources, then I can start doing some tests

**Comment by Toasterson** [comment]
Ok, this is now slowly ready to test. Probably not perfect but let's start testing on devices of ours.

**Comment by AndWac** [comment]
openindiana.org
  x11/library/mesa
    13.0.6-2024.0.0.9 -> 21.3.9-2024.0.0.0

pkg update: The following packages deliver conflicting action types to usr/include/GL/gl.h:

  link:
    pkg://openindiana.org/service/opengl/ogl-select@0.5.11,5.11-2020.0.1.5:20200330T163649Z
  file:
    pkg://openindiana.org/x11/library/mesa@21.3.9,5.11-2024.0.0.0:20240526T185943Z

These packages cannot be installed together. Any non-conflicting subset
of the above packages can be installed.

The following packages deliver conflicting action types to usr/include/GL/glext.h:

  link:
    pkg://openindiana.org/service/opengl/ogl-select@0.5.11,5.11-2020.0.1.5:20200330T163649Z
  file:
    pkg://openindiana.org/x11/library/mesa@21.3.9,5.11-2024.0.0.0:20240526T185943Z

These packages cannot be installed together. Any non-conflicting subset
of the above packages can be installed.

The following packages deliver conflicting action types to usr/include/GL/glx.h:

  link:
    pkg://openindiana.org/service/opengl/ogl-select@0.5.11,5.11-2020.0.1.5:20200330T163649Z
  file:
    pkg://openindiana.org/x11/library/mesa@21.3.9,5.11-2024.0.0.0:20240526T185943Z

These packages cannot be installed together. Any non-conflicting subset
of the above packages can be installed.

The following packages deliver conflicting action types to usr/include/GL/glxext.h:

  link:
    pkg://openindiana.org/service/opengl/ogl-select@0.5.11,5.11-2020.0.1.5:20200330T163649Z
  file:
    pkg://openindiana.org/x11/library/mesa@21.3.9,5.11-2024.0.0.0:20240526T185943Z

These packages cannot be installed together. Any non-conflicting subset
of the above packages can be installed.

**Comment by Toasterson** [comment]
Found a missing dependency while packaging on another machine

File: components/x11/mesa/manifests/sample-manifest.p5m [hunk 1]
```diff
@@ -52,73 +42,61 @@ file path=usr/include/GLES3/gl3ext.h
...
+file path=usr/lib/mesa/$(MACH64)/pkgconfig/osmesa.pc
+file path=usr/lib/mesa/$(MACH64)/pkgconfig/xatracker.pc
+link path=usr/lib/mesa/$(MACH64)/vdpau/libvdpau_r600.so \
+    target=libvdpau_r600.so.1.0.0
+link path=usr/lib/mesa/$(MACH64)/vdpau/libvdpau_r600.so.1 \
+    target=libvdpau_r600.so.1.0.0
+link path=usr/lib/mesa/$(MACH64)/vdpau/libvdpau_r600.so.1.0 \
+    target=libvdpau_r600.so.1.0.0
+file path=usr/lib/mesa/$(MACH64)/vdpau/libvdpau_r600.so.1.0.0
+hardlink path=usr/lib/xorg/modules/dri/i915_dri.so target=kms_swrast_dri.so
```
Comment (mtelka) [inline components/x11/mesa/manifests/sample-manifest.p5m:98]: Please use `PKG_HARDLINKS` to get deterministic sample manifest.

Comment (Toasterson) [reply components/x11/mesa/manifests/sample-manifest.p5m:98]: Fixed

**Comment by Toasterson** [comment]
thread local storage fixed now. People can test with https://dlc.openindiana.org/testing/mesa.p5p

**Pull request merged**